src/matlab/kineticsmethods.cpp

```diff
@@ -97,6 +97,9 @@ void kineticsmethods(int nlhs, mxArray* plhs[],
             case 16:
                 ok = kin_getRevRateConstants(kin,1,nr,h);
                 break;
+            case 17:
+                ok = kin_getDelta(kin,getInt(prhs[3]),nr,h);
```

Comment:
Why is it called just `kin_getDelta`?

Reply:
As referenced in the PR message, this is a pre-existing function in clib/ct.cpp that provides an interface to a range of 'delta' functions for reaction thermo - a single interface which provides access to deltaGibbs, deltaEntropy, etc., based on the 'job' number fed in as prhs[3].  As with other Matlab toolbox issues, rather than parsing the cost-benefit of this interface and the use of "magic numbers," I decided to punt and use as is, until the Matlab toolbox revamp.

Reply:
Right, reading comprehension is apparently not something I'm capable of today 🙄  I'm torn, then, between adding the `.m` files for the other `case`s in `kin_getDelta` and just leaving well enough alone. On the one hand, the files are basically trivial and potentially useful; on the other, "just do it in the rewrite that may or may not ever happen".

Reply:
As referenced in the PR message, `kin_getDelta` is a pre-existing function in ctlib/ct.cpp.  It accepts a job number (`prhs[3]`), and calculates a range of reaction thermo terms (`deltaGibbs`, `deltaEntropy`, and standard state versions of all three).

Rather than wade into the cost benefit of this "magic number" approach, as with other recent changes I'm going to go with the Matlab toolbox as is, and wait for the toolbox overhaul before making any broader edits to the Matlab interface...

Reply:
Yeah, it's basically trivial - I'm fine adding them.

Reply:
Did you mean to paste the same message again?

Another thought I had is that you're doubling up on the magic numbers in the `.m` file here by using `prhs[3]`. Why not just eliminate that argument (`0`) from the call in `getDeltaEnthalpies.m` and just specify the `0` in `kin_getDelta`?

Reply:
So the question is if we want to also expose those other `getDelta` functions.  If we do, your suggestion above would mean we need a separate magic number (`17`, `18`, etc.) for each of `getDeltaEnthalpies`, `getDeltaGibbs`, etc., at the `src/matlab/kineticsmethods.cpp` level.  In the original, there is just one magic number at this level, then a second magic number at the `kin_getDelta` level.

The latter is a little clearer in reflecting the structure of the code - they all call a common function, and pass a 2nd argument to that function.  But I don't know if that really matters, much.  The former has the benefit of exposing the user to fewer magic numbers, and also avoids the conflict `kineticsmethods.cpp` wherein `prhs[3]` refers to this flag in our case, but is also assigned to `irxn` on line 159.

I don't have a super-strong preference.  If forced to choose, I'd go with omitting the `0` from `getDeltaEnthalpies.m`.  But then my vote on adding the other 'delta' functions to to the matlab interface would probably be "no."

Reply:
As a counter-counter-point, all these kinds of kinetics functions are supposed to fit between job ids 10 and 19. Since its already at 17, there aren't enough "spaces" to add all 6 functions as each their own `case` in `kineticsmethods.cpp`. So that suggests using the one magic number (`17`) as the job id and the second magic number as the item to retrieve. Boy, this is awful.

> avoids the conflict `kineticsmethods.cpp` wherein `prhs[3]` refers to this flag in our case, but is also assigned to `irxn` on line 159.

Given the other ridiculosity in this interface, I feel like "overloading" `prhs[3]` is the least-confusion-causing thing about this 😂

Reply:
Lol.  Really, that `irxn` definition should be moved to line 49 - it only applies to `job < 10`.

I will move that, but otherwise stick with the current approach, in terms of `getDeltaEnthalpies.m` and the two magic numbers.